README.md

```diff
@@ -302,53 +264,65 @@ Start Apache httpd web server using
 /usr/sbin/apache2 -X -d ./cgi/apache2
 ```
 
-And in another shell call CGI script using curl
+To test the [CGI script](http://localhost:8080/cgi-bin/newtonraphson) we can not use the web browser, but need to use http client like [curl](https://curl.haxx.se/).
+A web browser uses the [GET http request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) and `text/html` as content type, but the CGI script requires a [POST http request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) and JSON string as request body.
+The curl command with a POST request can be run in another shell with
 
 ```shell
-curl --header "Content-Type: application/json" \
-  --request POST \
+curl --request POST \
   --data '{"guess":-20, "epsilon":0.001}' \
+  --header "Content-Type: application/json" \
   http://localhost:8080/cgi-bin/newtonraphson
 ```
 
 Should return the following JSON document as a response
 
 ```json
 {
-  "guess": -20,
   "root":-1.0000001181322415
 }
 ```
 
-The problem with CGI scripts is when the program does some initialization, you have to wait for it each visit. It is better to do the initialization once when the web service is starting up.
+Instead of curl we could use any http client in any language to consume the web service.
 
-## Web framework
+The problem with CGI scripts is when the program does some initialization, you have to wait for it each visit. It is
+better to do the initialization once when the web service is starting up.
 
-![flask](images/flask.svg.png "Flask")
+## Python web service
 
-A web framework is an abstraction layer for making web applications. It takes care of mapping a request on a certain url to a user defined function. And mapping the return of a user defined function to a response like an HTML page or an error message.
+![openapi](images/openapi.svg.png "OpenAPI")
 
-## Python
+| Pros | Cons |
+| --- | --- |
+| :heart: Python is great glue language | :no_entry: Pure Python is slower than C++ |
+| :heart: Web service discoverable and documented with OpenAPI specification | :no_entry: Exception thrown from C++ has number instead of message   |
```

Comment:
```suggestion
| :heart: Web service is easy to discover and nicely documented with OpenAPI specification | :no_entry: Exception thrown from C++ has number instead of message   |
```